### ISSUE TITLE
Namespaces with large numbers of PEPs are slow to load

### ISSUE THREAD

**Issue by nleroy917** [pepkit/pephub#72]
When a namespace has many PEPs it is slow to load as the server waits for the data to come from the database. This problem is especially evident for GEO PEPs (there are many). As such we can fix this with two changes:

1. The PEPs should be loaded asynchronously (via ajax).
2. The namespace page should support server-side pagination.

**Comment by nleroy917** [comment]
~Two views~ One view:
- **Default view:** server-side pagination should be done. Disable client-side pagination and eliminate search. Just used serverside pagination.
- ~**Optional view:** click a button to search through all of them to paginate and search through (wont work in the long term)~

Only do serverside search.

**Comment by nleroy917** [comment]
We agreed that the following could be returned:

```
{
"count:" 2500
"offset": 0
"limit:" 100
"items:" [ ... ]
}
```

**Comment by nleroy917** [comment]
So, we are using the `pepdbagent.Connection.get_namespace_info(namespace=namespace, user=user)` method to get the projection information for a specific namespace. It hangs for just a half-second while it loads the data...

We discussed using pagination, but this method doesn't support the `limit` and `offset` parameters so I don't think I can add that in yet. I'll open a corresponding issue on `pepdbagent`.

**Comment by nleroy917** [comment]
@nsheff @Khoroshevskyi  I've set up server-side pagination for the most part... it could maybe be improved still a bit and I'm sure there may be slight bugs, but it mirrors the GitHub pagination for a users repositories.

All of my changes are merged into `dev` if you want to pull them and check it out. You'll need a fresh virtual environment since this needs recent changes to `pepdbagent`:

```console
python -m venv .venv
pip install --upgrade pip && pip install -r requirements/requirements-all.txt
source .venv/bin/activate
```

Then run with `uvicorn pephub.main:app --reload` If you need a database connection, I can send you some env variables.

**Comment by khoroshevskyi** [comment]
@nleroy917 
One more feature that should be added to search and namespace + project page is:  

- [ ] Sort by functionality. Something like drop down list.

After this user can sort files based on: upload date, name or tag

I will work on this functionality in pepdbagent

**Comment by nleroy917** [comment]
Great idea!

**Comment by nleroy917** [comment]
This is working and should be solved with `v0.5.0` release #80
